libfreerdp/core/capabilities.c

```diff
@@ -3353,193 +3356,191 @@ static BOOL rdp_print_capability_sets(wStream* s, UINT16 numberCapabilities, BOO
 {
 	UINT16 type;
 	UINT16 length;
-	BYTE *bm, *em;
 
 	while (numberCapabilities > 0)
 	{
-		Stream_GetPointer(s, bm);
-		rdp_read_capability_set_header(s, &length, &type);
-		WLog_INFO(TAG, "%s ", receiving ? "Receiving" : "Sending");
-		em = bm + length;
+		size_t rest;
+		wStream sub;
+		if (!rdp_read_capability_set_header(s, &length, &type))
+			return FALSE;
 
-		if (Stream_GetRemainingLength(s) < (size_t)(length - 4))
-		{
-			WLog_ERR(TAG, "error processing stream");
+		WLog_INFO(TAG, "%s ", receiving ? "Receiving" : "Sending");
+		Stream_StaticInit(&sub, Stream_Pointer(s), length - 4);
```

Comment:
length should be checked (if it's less than 4)

Reply:
Sorry I've just noticed that  `rdp_read_capability_set_header` now does this check